projects/ledger-app/0001-token-admin-methods.md

```diff
@@ -0,0 +1,219 @@
+
+---
+
+When creating a custom token or NFT the transaction information also includes the name and symbol of the token, these should only exist when a transaction has the version byte 2.
+
+---
+
+> [!NOTE] Version field has been split into `signalBits` and `version`
+> The version field of the transaction was 2 bytes (first byte was reserved for later use) and since Ledger app version 1.1.0 the version field has become a 1 byte field and the first byte became the `signalBits`
+
+Version byte needs to be refactored to a 1 byte field and support for `signalBits` will not be added in this design so we will only allow `0x00` as `signalBits` for now.
+
+## Admin operations
+
+### Mint, Melt, Delegate and Destroy
+
+Sending a mint, melt, delegate or destroy transactions should use the same protocol as sending normal transactions.
+Mint and Melt involve destroying/creating custom token outputs and since the Ledger app does not have access to the input data it can only see the outputs, which means that a mint operation is a transaction with custom tokens outputs (possibly HTR outputs for change) and authority outputs, same for melt operations.
+
+Delegate is simpler since its actually just sending an authority output.
+
+Destroy operations usually do not even have authority outputs, so it actually is already supported.
+Granted that the token array will have the token uid of the authority being destroyed so the user still has to sign the token data even with no outputs of the token.
+
+### Create token
+
+Token creation transactions are special for a number of reasons:
+1. Transaction version byte is `0x03`
+2. Extra fields at the end of the transaction, namely `token_info_version`, `name` and `symbol`
+3. There will be an output with the created token, the token uid will not be on the tokens array since it does not exist yet
+	1. Token uid is the transaction id of the created transaction, which is created during mining.
+
+To deal with these we will create a new command to load the token data of the token being created.
+We will use the existing `SEND_TOKEN_DATA` for instruction code `0x08` and use the `P1` field to indicate which data is being sent.
+
+| Description                                 | P1   | P2   | CData                                                                                                                                                                   |
+| ------------------------------------------- | ---- | ---- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Send token data<br>for signing transactions | 0x00 | 0x00 | `version (1)` \|\|<br> `uid (32)` \|\|<br>`symbol_len (1)` \|\|<br>`symbol (symbol_len)` \|\|<br>`name_len (1)` \|\|<br>`name (name_len)` \|\|<br>`signature (32)`      |

```

Comment:
This command seems to be very specific. Why not send the transaction bytes with a desired type and let the Ledger app parse and handle it? This way the Ledger app would support more advanced use cases without any change.